census_blueprints/execute_sync.py

```diff
@@ -0,0 +1,121 @@
+        dest='invalid_threshold',
+        default=0,
+        required=False)
+    args = parser.parse_args()
+    return args
+
+
+def write_json_to_file(json_object, file_name):
+    with open(file_name, 'w') as f:
+        f.write(
+            json.dumps(
+                json_object,
+                ensure_ascii=False,
+                indent=4))
+
+
+def execute_job(
+        url,
+        folder_name='dbt-blueprint-logs',
+        file_name='job_details_response.json'):
+    print(f'Starting sync for {url.split("/")[-2]}')
+    job_run_req = execute_request.execute_request(
+        'POST', url, headers={})
+    job_run_response = json.loads(job_run_req.text)
+    execute_request.create_folder_if_dne(folder_name)
+    combined_name = execute_request.combine_folder_and_file_name(
+        folder_name, file_name)
+    write_json_to_file(
+        job_run_response, combined_name)
+    return job_run_response
+
+
+def main():
+    args = get_args()
+    invalid_threshold = int(args.invalid_threshold)
+    failure_threshold = int(args.failure_threshold)
+    url = args.url
+    check_status = execute_request.convert_to_boolean(args.check_status)
+
+    sync_id = url.split('/')[-2]
+    org_id = os.environ.get("SHIPYARD_ORG_ID") if os.environ.get(
+        'USER') == 'shipyard' else sync_id
+    fleet_log_id = os.environ.get("SHIPYARD_FLEET_LOG_ID") if os.environ.get(
+        'USER') == 'shipyard' else ''
+    log_id = os.environ.get("SHIPYARD_LOG_ID") if os.environ.get(
+        'USER') == 'shipyard' else ''
+    base_folder_name = execute_request.clean_folder_name(
+        f'census-blueprint-logs/{org_id}/{fleet_log_id}/{log_id}')
+
+    job_run_response = execute_job(
+        url,
+        folder_name=f'{base_folder_name}/responses',
+        file_name=f'sync_{sync_id}_response.json')
+
+    sync_run_id = job_run_response['data']['sync_run_id']
+    pickle_folder_name = execute_request.clean_folder_name(
+        f'census-blueprint-logs/{org_id}/{fleet_log_id}')
+    pickle_file_name = execute_request.combine_folder_and_file_name(
+        pickle_folder_name, 'sync_run_id.pickle')
+    with open(pickle_file_name, 'wb') as f:
+        pickle.dump(sync_run_id, f)
+
+    if check_status:
+        is_complete = False
+        # Census API returns null if you poll it too quickly the first time.
+        time.sleep(5)
+        while not is_complete:
+            sync_details_response = check_sync_status.get_sync_details(
+                url,
+                sync_run_id,
+                folder_name=f'{base_folder_name}/responses',
+                file_name=f'run_{sync_run_id}_response.json')
+            is_complete = sync_details_response['data'][
+                'status'] == 'completed' or sync_details_response['data']['status'] == 'failed'
+            if not is_complete:
+                print(
+                    f'Run {sync_run_id} is not complete. Waiting 30 seconds and trying again.')
+                time.sleep(30)
```

Comment:
Do we want a count check on this so it doesn't run indefinitely if something goes wrong or the sync takes a really long time.

Reply:
@ericelsken Do we want to include that arbitrary value in the code? Or would we rather have Shipyard handle that logic (kill job if it takes longer than X minutes) so that users can adjust as needed?

Reply:
I was asking just in general. I think from the script's perspective it makes sense to _not_ have the general check. Ideally you'd want the script to run as long as it needs. If you're running locally and needed it to stop, you ctrl-c it or something like that. 
With Shipyard and the guardrail, that should be the same type of thing. I guess I was asking if the script should have it in and of itself, which I guess not.